OneSchedule.Domain/NotificationScheduler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using OneSchedule.Domain.Abstractions;
+using OneSchedule.Entities;
+using Quartz.Impl;
+using Quartz.Lambda;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+
+namespace OneSchedule.Domain
+{
+    public class NotificationScheduler : INotificationScheduler
+    {
+        private readonly ITelegramBotClient _bot;
+
+        public NotificationScheduler(ITelegramBotClient bot)
+        {
+            _bot = bot;
+        }
+
+        public async void ScheduleNotifications(EventEntity eventEntity)
+        {
+
+            var scheduler = new StdSchedulerFactory().GetScheduler().Result;
+
+            foreach (var item in eventEntity.Notifications)
+            {
+                await scheduler.ScheduleJob(() => _bot.SendTextMessageAsync(eventEntity.ChatId, $"Event {eventEntity.Title} will begin at {eventEntity.StartDate}"),
```

Comment:
You can start all tasks at first and after that wait all of them using Task.WhenAll() method. It can speed up app because the app doesn't wait until each ScheduleJob method is completed in the loop